src/ontodia/data/rdf/rdfCacheableStore.ts

```diff
@@ -0,0 +1,261 @@
+            RDFS: PrefixFactory('http://www.w3.org/2000/01/rdf-schema#'),
+            FOAF: PrefixFactory('http://xmlns.com/foaf/0.1/'),
+            XSD: PrefixFactory('http://www.w3.org/2001/XMLSchema#'),
+            OWL: PrefixFactory('http://www.w3.org/2002/07/owl#'),
+        };
+    }
+
+    add(rdfGraph: RDFGraph, prefix?: string) {
+        this.rdfStorage.add(uniqueId(prefix) || (uniqueId(DEFAULT_STOREG_URI)), rdfGraph);
+        this.enrichMaps(rdfGraph);
+    }
+
+    match(
+        subject?: string,
+        predicate?: string,
+        object?: string,
+        iri?: string,
+        callback?: (...args: any[]) => void,
+        limit?: number,
+    ): Promise<RDFGraph> {
+        if (subject && (LABEL_URIS.indexOf(predicate) !== -1) && !object) {
+            return Promise.resolve(this._getLabels(subject));
+        } else if (subject && predicate === this.prefs.RDF('type') && !object) {
+            return Promise.resolve(this.getTypes(subject));
+        } else {
+            return this.rdfStorage.match(
+                subject,
+                predicate,
+                object,
+                iri,
+                callback,
+                limit,
+            );
+        }
+    }
+
+    matchAll(statements: MatchStatement[]): Promise<RDFGraph> {
+        const slowQueries: MatchStatement[] = [];
+        const queries: RDFGraph[] = [];
+
+        statements.forEach(statement => {
+            if (statement.subject && (LABEL_URIS.indexOf(statement.predicate) !== -1) && !statement.object) {
+                queries.push(this._getLabels(statement.subject));
+            } else if (statement.subject && statement.predicate === this.prefs.RDF('type') && !statement.object) {
+                queries.push(this.getTypes(statement.subject));
+            } else {
+                slowQueries.push(statement);
+            }
+        });
+
+        queries.push(this.multipleMatch(slowQueries));
+
+        return Promise.resolve(this.combineGraphs(queries));
+    }
+
+    getLabels(id: string): Promise<RDFGraph> {
+        return Promise.resolve(createGraph(this.labelsMap[id]));
+    }
+
+    // Checks whetger the element is in the storage.
+    checkElement(id: string): Promise<boolean> {
+        if (this.labelsMap[id]) { // if there is label for an id then the element is already fetched
+            return Promise.resolve(true);
+        } else {
+            if (!this.checkingElementMap[id]) {
+                this.checkingElementMap[id] = this.rdfStorage.match(id, null, null).then(result => {
+                    const resultArray = result.toArray();
+                    if (resultArray.length === 0) {
+                        return false;
+                    } else {
+                        return true;
+                    }
+                });
+                return this.checkingElementMap[id];
+            } else {
+                return this.checkingElementMap[id];
```

Comment:
This line is duplicated in both `if`-`else` branches, so you could put it outside after conditional statement.